src/routes/device-type/update.ts

```diff
@@ -18,7 +18,22 @@ interface Request {
       name: string;
       pinFieldConfigurations: UpdatableFieldConfigurations;
       defaultPinName?: string;
+      charts?: {
+        title: string | null;
+        series: {
+          quantityHashId: string;
+          color: string;
+        }[];
+      }[];
     }[];
+    charts?: {
+      title: string | null;
+      series: {
+        channelIndex: number;
+        quantityHashId: string;
+        color: string;
+      }[];
+    }[] | null;
```

Comment:
null is not needed I think (see above)

Reply:
Both charts objects in deviceType.add should be optional for backwards compatibility (the one on root-level already is optional, so that's good!) .